src/BenchmarkDotNet.Core/Exporters/RPlotExporter.cs

```diff
@@ -36,33 +37,54 @@ public IEnumerable<string> ExportToFiles(Summary summary)
             lock (buildScriptLock)
                 File.WriteAllText(scriptFullPath, script);
 
-            // TODO: implement smart autodetection of the R bin folder
+            string rscriptPath;
             var rHome = Environment.GetEnvironmentVariable("R_HOME");
-            if (Directory.Exists(rHome))
+            if (rHome != null)
             {
-                var start = new ProcessStartInfo
+                rscriptPath = Path.Combine(rHome, "Rscript.exe");
+                if (!File.Exists(rscriptPath))
                 {
-                    UseShellExecute = false,
-                    RedirectStandardOutput = false,
-                    CreateNoWindow = true,
-                    FileName = Path.Combine(rHome, "Rscript.exe"),
-                    WorkingDirectory = summary.ResultsDirectoryPath,
-                    Arguments = $"\"{scriptFullPath}\" \"{fileNamePrefix}-measurements.csv\""
-                };
-                using (var process = Process.Start(start))
-                    process?.WaitForExit();
-                yield return fileNamePrefix + "-boxplot.png";
-                yield return fileNamePrefix + "-barplot.png";
+                    rscriptPath = Path.Combine(rHome, "Rscript.exe");
+                    if (!File.Exists(rscriptPath))
+                    {
+                        consoleLogger.WriteLineError($"RPlotExporter requires R_HOME to point to the directory containing Rscript.exe (currently points to {rHome})");
+                        yield break;
+                    }
+                }
             }
-            else
+            else // No R_HOME, try the path
```

Comment:
You could wrap each branch of the `if` in a separated method with a nice name (something like `TryFindRScriptInRHome` and `TryFindRScriptInPath`), it will make code simpler.